flax/core/scope.py

```diff
@@ -761,6 +765,13 @@ def _populate_collections(self):
     for col in collections:
       self._collection(col)
 
+  def has_flag(self, key) -> bool:
+    return key in self.flags
+
+  def get_flag(self, key) -> Any:
```

Comment:
Maybe `get_flag` could accept a default in case flag doesn't exist?

```python
  def get_flag(self, key, default=None) -> Any:
    if key not in self.flags:
      if default is None:
        return ValueError(f'Flag {key} not present in scope.')
      return default
    return self.flags[key]
```

Reply:
I added an optional default argument - if none provided it throws an error on missing flag.